x-pack/plugins/secops/public/containers/kuery_autocompletion/index.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { AutocompleteSuggestion, getAutocompleteProvider } from 'ui/autocomplete_providers';
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+type RendererResult = React.ReactElement<JSX.Element> | null;
+type RendererFunction<RenderArgs, Result = RendererResult> = (args: RenderArgs) => Result;
+
+interface KueryAutocompletionLifecycleProps {
+  children: RendererFunction<{
+    isLoadingSuggestions: boolean;
+    loadSuggestions: (expression: string, cursorPosition: number, maxSuggestions?: number) => void;
+    suggestions: AutocompleteSuggestion[];
+  }>;
+  indexPattern: StaticIndexPattern;
+}
+
+interface KueryAutocompletionLifecycleState {
+  // lacking cancellation support in the autocompletion api,
+  // this is used to keep older, slower requests from clobbering newer ones
+  currentRequest: {
+    expression: string;
+    cursorPosition: number;
+  } | null;
```

Comment:
Optional: I would make this an exported Type called `CurrentRequest` above this interface to make this read a bit better?